packages/api-sdk/src/medical/client/__tests__/metriport.test.e2e.ts

```diff
@@ -0,0 +1,149 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+import dayjs from "dayjs";
+import { v4 as uuidv4 } from "uuid";
+import { getEnvVar, getEnvVarOrFail, ISO_DATE } from "../../../shared";
+import { MetriportMedicalApi } from "../metriport";
+
+const apiKey = getEnvVarOrFail("TEST_API_KEY");
+const baseAddress = getEnvVarOrFail("API_URL");
+const patientId = getEnvVar("TEST_PATIENT_ID");
+
+if (!patientId) {
+  console.log(`Missing TEST_PATIENT_ID env var, SKIPPING E2E TESTS!!!`);
+  it.skip("skipping e2e tests", () => {});
+} else {
+  const metriport = new MetriportMedicalApi(apiKey, { baseAddress });
+
+  describe("getMetriportUserId", () => {
+    it("returns documents when no filters", async () => {
+      const { documents } = await metriport.listDocuments(patientId);
+      expect(documents).toBeTruthy();
+      expect(documents.length).toBeGreaterThanOrEqual(1);
+    });
+
+    it("returns no documents when dateFrom is future", async () => {
+      const filters = { dateFrom: dayjs().add(3, "day").format(ISO_DATE) };
+      const { documents } = await metriport.listDocuments(patientId, filters);
+      expect(documents).toBeTruthy();
+      expect(documents.length).toEqual(0);
+    });
+
+    it("returns no documents when dateTo is ancient", async () => {
+      const filters = {
+        dateTo: dayjs().subtract(100, "years").format(ISO_DATE),
+      };
+      const { documents } = await metriport.listDocuments(patientId, filters);
+      expect(documents).toBeTruthy();
```

Comment:
For my own learning, in cases like this, is it really necessary to be checking if `documents` is truthy every time? Wouldn't it be sufficient to do that in the first test of this kind? Especially if the function is always expected to return an array (that could sometimes be empty)?

Reply:
Its kinda checking whether the value is present or not. The function is expected to return an array, but its a SDK and its not doing schema validation on the response, so could in theory return null/undefined.

Some permutations could save the check for truthy, but since it would require additional cycles to decide which ones, I reckoned its easier to just leave it there - so tests won't fail b/c `undefined does't have property x` - and also b/c its super simple, so doesn't add much to the test